interview-questions.md

```diff
@@ -8,50 +8,55 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: I know that the arguments are the variable refered to when you invoke a function that intakes parameters. Too my knowledge you can use parameter and arguments interchangably.
 
-Researched answer:
+Researched answer: The difference between parameters and an argument is that the argument is refered to the vaiables when a function is invoked but parameter is refering to the variables when the function is created/declared.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: Value(the value you of the element) Index(the index of the value), array(the array itself) - value is the required parameter. index and array are optional.
 
-Researched answer:
+Researched answer: "currentValue" this is the only requireed parameter. it contains the value of the current element. Index, this contains the index of the current element and this one is optional. Array, this contains the array on which the map function is applied.
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: map and filter both iterate through an array. they both return a new array; map returns the same size array but filter will return a subset of the prior array. Both map and filter area array prototype which means they both must be called on an array.
 
-Researched answer:
+Researched answer: Both map and filter are prototypes arrays which means they oth must be called upon an array. With that being said there are some differences and scenarios where we should use map over filter or filter over map; First off, map iterates through the array value by value and performs a specfic operation over each value and will return a new array BUT WITH the same length. Filter on the other hand, will create a new array but a subset of the prior array if the value makes it through the conditional statement within the filter.
 
 1. What is iteration?
 
-Your answer:
+Your answer: Iteration is the process in which we perform count in a specific manner until a conditions is met. For an example - for loops; you can specify where the iteration starts and counting by a specific amount until we reach a certain point
 
-Researched answer:
+Researched answer:iteration is simply the number of time(s) a loop can be executed before a condition is met and therefore stops the iteration.
```

Comment:
Iteration can also occur without a condition being met. Can you think of an example?